File: Musixmatch_lyrics_API_App/app/src/main/java/alexejantonov/com/musixmatch_lyrics_api_app/ui/seach_screen/SearchFragmentView.java
package alexejantonov.com.musixmatch_lyrics_api_app.ui.seach_screen;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

import java.util.List;

import alexejantonov.com.musixmatch_lyrics_api_app.ui.Base.BaseData;

@StateStrategyType(AddToEndSingleStrategy.class)
public interface SearchFragmentView extends MvpView {
	void showData(List<BaseData> data);
}

File: Musixmatch_lyrics_API_App/app/src/main/java/alexejantonov/com/musixmatch_lyrics_api_app/ui/seach_screen/SearchFragmentView.kt
package alexejantonov.com.musixmatch_lyrics_api_app.ui.seach_screen

import alexejantonov.com.musixmatch_lyrics_api_app.ui.Base.BaseData
import com.arellomobile.mvp.MvpView
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType

@StateStrategyType(AddToEndSingleStrategy::class)
interface SearchFragmentView : MvpView {
  fun showData(data: List<BaseData>, query: String)
}
